src/pixel.rs

```diff
@@ -0,0 +1,146 @@
+/// last unset), except on WebAssembly targets (`target_family = "wasm"`), there it is RGBX, since
+/// the API on the Web only supports that format. This distinction should only be relevant if you're
+/// bitcasting `Pixel` to/from a `u32`.
+///
+/// If you're familiar with [the `rgb` crate](https://docs.rs/rgb/), you can treat this mostly as-if
+/// it was defined as follows:
+///
+/// ```ignore
+/// #[cfg(target_family = "wasm")]
+/// type Pixel = rgb::Rgba<u8>;
+/// #[cfg(not(target_family = "wasm"))]
+/// type Pixel = rgb::Bgra<u8>;
+/// ```
+///
+/// # Example
+///
+/// Construct a new pixel.
+///
+/// ```
+/// # use softbuffer::Pixel;
+/// #
+/// let red = Pixel::new_rgb(0xff, 0x80, 0);
+/// assert_eq!(red.r, 255);
+/// assert_eq!(red.g, 128);
+/// assert_eq!(red.b, 0);
+/// assert_eq!(red.a, 0xff);
+///
+/// let from_struct_literal = Pixel { r: 255, g: 0x80, b: 0, a: 0xff };
+/// assert_eq!(red, from_struct_literal);
+/// ```
+///
+/// Convert a pixel to an array of `u8`s.
+///
+/// ```
+/// # use softbuffer::Pixel;
+/// #
+/// let red = Pixel::new_rgb(0xff, 0, 0);
+/// // SAFETY: `Pixel` can be reinterpreted as `[u8; 4]`.
+/// let red = unsafe { core::mem::transmute::<Pixel, [u8; 4]>(red) };
+///
+/// if cfg!(target_family = "wasm") {
+///     // RGBX
+///     assert_eq!(red[0], 255);
+/// } else {
+///     // BGRX
+///     assert_eq!(red[2], 255);
+/// }
+/// ```
+///
+/// Convert a pixel to an `u32`.
+///
+/// ```
+/// # use softbuffer::Pixel;
+/// #
+/// let red = Pixel::new_rgb(0xff, 0, 0);
+/// // SAFETY: `Pixel` can be reinterpreted as `u32`.
+/// let red = unsafe { core::mem::transmute::<Pixel, u32>(red) };
+///
+/// if cfg!(target_family = "wasm") {
+///     // RGBX
+///     assert_eq!(red, u32::from_ne_bytes([0xff, 0x00, 0x00, 0xff]));
+/// } else {
+///     // BGRX
+///     assert_eq!(red, u32::from_ne_bytes([0x00, 0x00, 0xff, 0xff]));
+/// }
+/// ```
+#[repr(C)]
+#[repr(align(4))] // May help the compiler to see that this is a u32
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct Pixel {
+    #[cfg(target_family = "wasm")]
+    /// The red component.
+    pub r: u8,
+    #[cfg(target_family = "wasm")]
+    /// The green component.
+    pub g: u8,
+    #[cfg(target_family = "wasm")]
+    /// The blue component.
+    pub b: u8,
+
+    #[cfg(not(target_family = "wasm"))]
+    /// The blue component.
+    pub b: u8,
+    #[cfg(not(target_family = "wasm"))]
+    /// The green component.
+    pub g: u8,
+    #[cfg(not(target_family = "wasm"))]
+    /// The red component.
+    pub r: u8,
+
+    /// The alpha component.
+    ///
+    /// `0xff` here means opaque, whereas `0` means transparent.
+    ///
+    /// NOTE: Transparency is yet poorly supported, see [#17], until that is resolved, you will
+    /// probably want to set this to `0xff`.
+    ///
+    /// [#17]: https://github.com/rust-windowing/softbuffer/issues/17
+    pub a: u8,
```

Comment:
We _could_ leave this out, but:
- There may be backends that currently read this, and leaving it as padding in the struct might cause them to read uninitialized bytes.
- It'll be useful in moving towards https://github.com/rust-windowing/softbuffer/issues/17 with less of a breaking change.